### PR TITLE
Bump KafkaChannel types to v1beta1

### LINKED ISSUE #1351: KafkaChannel v1beta1 type bumps
Part of https://github.com/knative/eventing-contrib/issues/1350

### PR DESCRIPTION
**Pull request by aliok**
Fixes #1351

## Proposed Changes

- v1beta1 type bump for KafkaChannel
- 
-

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

File: kafka/channel/pkg/apis/messaging/v1beta1/kafka_channel_defaults.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"context"
```
Comment (mattmoor) [inline kafka/channel/pkg/apis/messaging/v1beta1/kafka_channel_defaults.go:20]: Format Go code:
```suggestion
	"context"

```

File: kafka/channel/pkg/apis/messaging/v1beta1/kafka_channel_defaults_test.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"context"
```
Comment (mattmoor) [inline kafka/channel/pkg/apis/messaging/v1beta1/kafka_channel_defaults_test.go:17]: Format Go code:
```suggestion
	"context"
	"testing"

```

File: kafka/channel/pkg/apis/messaging/v1beta1/kafka_channel_defaults_test.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
```
Comment (mattmoor) [inline kafka/channel/pkg/apis/messaging/v1beta1/kafka_channel_defaults_test.go:19]: Format Go code:
```suggestion

```

File: kafka/channel/pkg/apis/messaging/v1beta1/kafka_channel_defaults.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+	// we support. Reason for this is that the stored version will
+	// not get a chance to get modified, but for newer versions
+	// conversion webhook will be able to take a crack at it and
+	// can modify it to match the duck shape.
+	if c.Annotations == nil {
+		c.Annotations = make(map[string]string)
+	}
+	if _, ok := c.Annotations[messaging.SubscribableDuckVersionAnnotation]; !ok {
+		c.Annotations[messaging.SubscribableDuckVersionAnnotation] = "v1"
+	}
```
Comment (aliok) [inline kafka/channel/pkg/apis/messaging/v1beta1/kafka_channel_defaults.go:38]: We need this to support duckv1, right?

File: kafka/channel/pkg/apis/messaging/v1beta1/kafka_channel_types.go [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+// KafkaChannelSpec defines the specification for a KafkaChannel.
+type KafkaChannelSpec struct {
+	// NumPartitions is the number of partitions of a Kafka topic. By default, it is set to 1.
+	NumPartitions int32 `json:"numPartitions"`
+
+	// ReplicationFactor is the replication factor of a Kafka topic. By default, it is set to 1.
+	ReplicationFactor int16 `json:"replicationFactor"`
+
+	// KafkaChannel conforms to Duck type Subscribable.
+	Subscribable *eventingduck.Subscribable `json:"subscribable,omitempty"`
```
Comment (aliok) [inline kafka/channel/pkg/apis/messaging/v1beta1/kafka_channel_types.go:71]: Should I change this line with 
```
eventingduckv1.ChannelableSpec `json:",inline"`
```
similar to InMemoryChannel?

We'd also have a `Delivery *DeliverySpec` in that case in `KafkaChannelSpec`

More context: https://knative.slack.com/archives/CR2141CGY/p1593428283210700

Comment (aliok) [reply kafka/channel/pkg/apis/messaging/v1beta1/kafka_channel_types.go:71]: cc @n3wscott

**Comment by aliok** [comment]
cc @lionelvillard @matzew
Please ignore the failing coverage

File: kafka/channel/pkg/apis/messaging/v1beta1/kafka_channel_defaults.go (hunk 4 above)
Comment (matzew) [reply kafka/channel/pkg/apis/messaging/v1beta1/kafka_channel_defaults.go:38]: yep

File: kafka/channel/pkg/apis/messaging/v1beta1/kafka_channel_types.go (hunk 5 above)
Comment (vaikas) [reply kafka/channel/pkg/apis/messaging/v1beta1/kafka_channel_types.go:71]: Yea and also the Status

Comment (matzew) [reply kafka/channel/pkg/apis/messaging/v1beta1/kafka_channel_types.go:71]: @aliok  something like:

```yaml

// KafkaChannelSpec defines the specification for a KafkaChannel.
type KafkaChannelSpec struct {
	// NumPartitions is the number of partitions of a Kafka topic. By default, it is set to 1.
	NumPartitions int32 `json:"numPartitions"`

	// ReplicationFactor is the replication factor of a Kafka topic. By default, it is set to 1.
	ReplicationFactor int16 `json:"replicationFactor"`

	// Channel conforms to Duck type Channelable.
	eventingduck.ChannelableSpec `json:",inline"`
}

// KafkaChannelStatus represents the current state of a KafkaChannel.
type KafkaChannelStatus struct {
	// Channel conforms to Duck type Channelable.
	eventingduck.ChannelableStatus `json:",inline"`
}
```

**Comment by matzew** [comment]
/lgtm
/approve

I will follow up w/ a PR for the types

**Pull request merged**